compiler/rustc_borrowck/src/lib.rs

```diff
@@ -444,13 +443,15 @@ fn do_mir_borrowck<'tcx>(
         tainted_by_errors,
     };
 
-    let body_with_facts = if return_body_with_facts {
-        let output_facts = mbcx.polonius_output.expect("Polonius output was not computed");
+    let body_with_facts = if consumer_options.is_some() {
+        let output_facts = mbcx.polonius_output;
         Some(Box::new(BodyWithBorrowckFacts {
             body: body_owned,
-            input_facts: *polonius_input.expect("Polonius input facts were not generated"),
+            borrow_set,
+            region_inference_context: regioncx,
+            location_table: polonius_input.as_ref().map(|_| location_table_owned),
+            input_facts: polonius_input,
```

Comment:
These two fields are always `Some` or `None` together, it may make sense to merge them into a common struct and put that into the `Option`

Reply:
I agree but I can't think of a reasonable name to give to that struct.

Reply:
yea... it's not clear what's best to me either. Especially since there's the output facts that partially overlap, too.